envoy/api/v2/core/address.proto

```diff
@@ -71,6 +85,9 @@ message BindConfig {
   // flag is not set (default), the socket is not modified, i.e. the option is
   // neither enabled nor disabled.
   google.protobuf.BoolValue freebind = 2;
+
+  // If set then set SO_KEEPALIVE on the socket to enable TCP Keepalives.
+  TcpKeepalive tcp_keepalive = 3;
```

Comment:
Is this tied to the prebind configuration? Or can it be set at any time (e.g. post bind, after `listen()`, etc)?

Reply:
My current implementation is only applicable to upstream connections, and is therefore prebind since ClientConnectionImpl only invokes PreBind.
Making it usable for inbound connections will be a bit more work, since I believe the options would need to be set on the newly accept()ed socket rather than the listen() socket, and there's no existing pattern for doing that.

Reply:
I still have a uncomfortable feeling about this grouping. So far, `BindConfig` is all the socket address binding config, this is more of a general connection property. Can we make this a new `ConnectionOptions` message that belongs to Cluster?

Reply:
FWIW, I'm going to eventually add keepalive settings on listeners too. So having these in a separate message makes sense to me. But we could put this message into a ConnectionOptions.